parachain/runtime/runtime-common/src/tests.rs

```diff
@@ -0,0 +1,246 @@
+	) -> Result<Assets, XcmError> {
+		Ok(Assets::default())
+	}
+}
+
+pub struct NotFoundTransactor;
+impl TransactAsset for NotFoundTransactor {
+	fn can_check_in(
+		_origin: &MultiLocation,
+		_what: &MultiAsset,
+		_context: &XcmContext,
+	) -> XcmResult {
+		Err(XcmError::AssetNotFound)
+	}
+
+	fn can_check_out(
+		_dest: &MultiLocation,
+		_what: &MultiAsset,
+		_context: &XcmContext,
+	) -> XcmResult {
+		Err(XcmError::AssetNotFound)
+	}
+
+	fn deposit_asset(
+		_what: &MultiAsset,
+		_who: &MultiLocation,
+		_context: Option<&XcmContext>,
+	) -> XcmResult {
+		Err(XcmError::AssetNotFound)
+	}
+
+	fn withdraw_asset(
+		_what: &MultiAsset,
+		_who: &MultiLocation,
+		_context: Option<&XcmContext>,
+	) -> Result<Assets, XcmError> {
+		Err(XcmError::AssetNotFound)
+	}
+
+	fn internal_transfer_asset(
+		_what: &MultiAsset,
+		_from: &MultiLocation,
+		_to: &MultiLocation,
+		_context: &XcmContext,
+	) -> Result<Assets, XcmError> {
+		Err(XcmError::AssetNotFound)
+	}
+}
+
+#[test]
+fn handle_fee_success() {
+	let fee: MultiAssets = MultiAsset::from((MultiLocation::parent(), 10_u128)).into();
+	let ctx = XcmContext {
+		origin: Some(MultiLocation { parents: 1, interior: X1(Parachain(1000)) }),
+		message_id: XcmHash::default(),
+		topic: None,
+	};
+	let reason = FeeReason::Export { network: EthereumNetwork::get(), destination: Here };
+	let result = XcmExportFeeToSibling::<
+		u128,
+		u64,
+		TokenLocation,
+		EthereumNetwork,
+		SuccessfulTransactor,
+		MockOkOutboundQueue,
+	>::handle_fee(fee, Some(&ctx), reason);
+	let local_fee =
+		MultiAsset::from((MultiLocation::parent(), MockOkOutboundQueue::local_fee())).into();
+	// assert only local fee left
+	assert_eq!(result, local_fee)
+}
+
+#[test]
+fn handle_fee_success_but_not_for_ethereum() {
+	let fee: MultiAssets = MultiAsset::from((MultiLocation::parent(), 10_u128)).into();
+	let ctx = XcmContext { origin: None, message_id: XcmHash::default(), topic: None };
+	// invalid network not for ethereum
+	let reason = FeeReason::Export { network: Kusama, destination: Here };
+	let result = XcmExportFeeToSibling::<
+		u128,
+		u64,
+		TokenLocation,
+		EthereumNetwork,
+		SuccessfulTransactor,
+		MockOkOutboundQueue,
+	>::handle_fee(fee.clone(), Some(&ctx), reason);
+	// assert fee not touched and just forward to the next handler
+	assert_eq!(result, fee)
+}
+
+#[test]
+fn handle_fee_fail_for_invalid_location() {
+	let fee: MultiAssets = MultiAsset::from((MultiLocation::parent(), 10_u128)).into();
+	// invalid origin not from sibling chain
+	let ctx = XcmContext { origin: None, message_id: XcmHash::default(), topic: None };
+	let reason = FeeReason::Export { network: EthereumNetwork::get(), destination: Here };
+	let result = XcmExportFeeToSibling::<
+		u128,
+		u64,
+		TokenLocation,
+		EthereumNetwork,
+		SuccessfulTransactor,
+		MockOkOutboundQueue,
+	>::handle_fee(fee.clone(), Some(&ctx), reason);
+	assert_eq!(result, fee)
+}
+
+#[test]
+fn handle_fee_fail_for_fees_not_met() {
```

Comment:
Needs a comment explaining what condition is being tested.

Reply:
https://github.com/Snowfork/snowbridge/pull/1109/commits/18daf99473b759c8c278815dcf33bd4e75616a7a